ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java

```diff
@@ -409,20 +412,29 @@
     }
 
     final StoreTransaction transaction = recentChainData.startStoreTransaction();
-    final UInt64 earliestAffectedSlot =
-        recentChainData
-            .getSlotForBlockRoot(block.getParentRoot())
-            .map(UInt64::increment)
-            .orElse(block.getSlot());
     addParentStateRoots(spec, blockSlotState, transaction);
+
+    final List<BlobSidecar> blobSidecars;
+    if (blobSidecarsAndValidationResult.isNotRequired()) {
+      // Outside availability window or pre-Deneb
+      blobSidecars = Collections.emptyList();
+    } else if (blobSidecarsAndValidationResult.isValid()) {
+      blobSidecars = blobSidecarsAndValidationResult.getBlobSidecars();
+    } else {
+      throw new IllegalStateException(
+          String.format(
+              "Unexpected attempt to store invalid blob sidecars (%s) for block: %s",
+              blobSidecarsAndValidationResult, block));
```

Comment:
I think codescan is right here. Doesn't make sense to print `blobSidecarsAndValidationResult`. Maybe number of blobs? For block I'd use `toLogString`